src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -361,11 +363,17 @@
                                     mq-endpoint
                                     (command-names command) version payload)))
 
-(defn -main
+(defn- main
   "Calls the trapperkeeper main argument to initialize tk.
 
   For configuration customization, we intercept the call to parse-config-data
   within TK."
   [& args]
   (rh/add-hook #'puppetlabs.trapperkeeper.config/parse-config-data #'conf/hook-tk-parse-config-data)
-  (apply main args))
+  (try+
+   (apply tk/main args)
+   (catch [:type ::conf/configuration-error] obj
+     (log/error (:message obj))
+     (throw+ (assoc obj ::utils/exit-status 1)))))
+
+(def -main (utils/wrap-main main))
```

Comment:
oh, I see, you wrapped it with this one. That's rather confusing... can the above one be called something different?